pkgs/tools/misc/ethminer/default.nix

```diff
@@ -0,0 +1,62 @@
+{ fetchurl, lib, pkgs, stdenv, fetchFromGitHub, opencl-headers, cmake, jsoncpp, boost, makeWrapper,
+  cudatoolkit, mesa, ethash, opencl-info, ocl-icd, openssl, pkg-config, cli11 }:
+
+stdenv.mkDerivation rec {
+  name = "${pkgname}-${version}";
+  pkgname = "ethminer";
+  version = "0.18.0-alpha.3";
+
+  src =
+    fetchFromGitHub {
+      owner = "ethereum-mining";
+      repo = "ethminer";
+      rev = "v${version}";
+      sha256 = "0kcgp5kiir8zncmjfr36nlqs363hm8d2m9fc9ld88bjvqwilz7bz";
+      fetchSubmodules = true;
+    };
+
+  # NOTE: dbus is broken
+  cmakeFlags = [
+    "-DHUNTER_ENABLED=OFF"
+    "-DETHASHCUDA=ON"
+    "-DAPICORE=ON"
+    "-DETHDBUS=OFF"
+    "-DCMAKE_BUILD_TYPE=Release"
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    makeWrapper
+  ];
+
+  buildInputs = [
+    cli11
+    boost
+    opencl-headers
+    mesa
+    cudatoolkit
+    ethash
+    opencl-info
+    ocl-icd
+    openssl
+    jsoncpp
+  ];
+
+  preConfigure = ''
+    sed -i 's/_lib_static//' libpoolprotocols/CMakeLists.txt
+  '';
+
+  postInstall = ''
+    wrapProgram $out/bin/ethminer --set LD_LIBRARY_PATH "${pkgs.linuxPackages.nvidia_x11}/lib"
```

Comment:
If you need to get it working properly with opengl drivers, you should wrap it with `--prefix LD_LIBRARY_PATH : /run/opengl-driver/lib`. This here would only work for a specific nvidia driver.